### PR TITLE
Correctly set TARGET variables when building for Android.

### PR DESCRIPTION
**Pull request by pietro**
With this change I was able to compile ring using the Android SDK/NDK. I was able to build and run the tests using the docker container rust libc uses to test on android. Un fortunately the tests segfault like some of the other arm builds. I'll turn my android experimentation into a script and do a PR to add android to travis later.

### CODE REVIEW

**Comment by pietro** [comment]
Well, back to reading about make conditionals. Also a similar change is needed to build on ios. I'll try to get my hands on a mac to test if setting the ABI to "macho" works.

File: mk/top_of_makefile.mk [hunk 1]
```diff
@@ -27,10 +27,12 @@ TARGET_VENDOR = $(word 2,$(TARGET_WORDS))
 TARGET_SYS = $(word 3,$(TARGET_WORDS))
 TARGET_ABI = $(word 4,$(TARGET_WORDS))
 
-# Cargo doesn't pass the ABI as part of TARGET on Mac OS X.
+# Cargo doesn't pass the ABI as part of TARGET on Mac OS X or Android.
```
Comment (briansmith) [inline mk/top_of_makefile.mk]: If I understand your patch correctly, for Android it omits the *vendor* part, not the ABI. So, this comment should be updated.

File: mk/top_of_makefile.mk [hunk 2]
```diff
@@ -27,10 +27,12 @@ TARGET_VENDOR = $(word 2,$(TARGET_WORDS))
...
 ifeq ($(TARGET_ABI),)
 ifeq ($(findstring apple-darwin,$(TARGET_VENDOR)-$(TARGET_SYS)),apple-darwin)
 TARGET_ABI = macho
+ifeq ($(findstring linux-androideabi,$(TARGET_VENDOR)-$(TARGET_SYS)),linux-androideabi)
+TARGET_ABI = androideabi
```
Comment (briansmith) [inline mk/top_of_makefile.mk]: We should set `TARGET_SYS` to `linux`. Maybe `TARGET_VENDOR` to `unknown`?

File: mk/top_of_makefile.mk [hunk 3]
```diff
@@ -39,14 +41,16 @@ endef
 $(error TARGET must be of the form \
         <arch>[<sub>]-<vendor>-<sys>-<abi>.$(NEWLINE)\
 \
-\       Exceptions: <abi> defaults to "macho" on Mac OS X.\
-\
+        Exceptions: <abi> defaults to "macho" on Mac OS X \
+        and to "androideabi" on Android. $(NEWLINE)\
```
Comment (briansmith) [inline mk/top_of_makefile.mk]: Again, `<vendor>` defaults to `unknown` on Android, and the ABI is explicitly given.

**Comment by briansmith** [comment]
Please add this to your commit messages:
```
<blank line>
I agree to license my contributions to each file under the terms given at the top of each file I changed.
```

Thanks!

File: mk/top_of_makefile.mk (hunk 1 above)
Comment (pietro) [inline mk/top_of_makefile.mk]: Yeah, you're right and it's not exactly cargo, but how the rust team named the platforms it builds for. I'm working on the wording to explain this.

**Comment by briansmith** [comment]
Awesome! This landed as d73194733209ffcc3704b79d71deee5375d6662b. Note that I wrapped the last line of the commit message at 72 characters.

Thanks!

**Pull request closed without merging**